src/ObjectStore/v1/Service.php

```diff
@@ -86,4 +86,42 @@ public function containerExists(string $name): bool
             throw $e;
         }
     }
+
+    /**
+     * Creates a temporary URL to access object in private containers.
+     * This method loosely follows swift command's way to generate temporary url: `swift tempurl $METHOD $EXPIRE $PATH $KEY`.
+     *
+     * @param string $method  An HTTP method to allow for this temporary URL. Any of GET, POST, HEAD, PUT, POST, DELETE.
+     * @param int    $expires Unix timestamp
+     * @param string $path    The full path or storage URL to the Swift object. Example: '/v1/AUTH_account/c/o'  or: 'http://saio:8080/v1/AUTH_account/c/o'
+     *                        For prefix based signature, set path to 'prefix:/v1/AUTH_account/container/pre'
+     * @param string $key     The secret temporary URL key set on the Swift cluster*
+     * @param string $ipRange [OPTIONAL] If present, the temporary URL will be restricted to the given ip or ip range
+     * @param string $digest  [OPTIONAL] The digest algorithm to be used may be configured by the operator. Default to sha1.
+     *                        Check the tempurl.allowed_digests  entry in the cluster's capabilities response to see which algorithms are supported by your
+     *                        deployment;
+     *
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    public function tempUrl(string $method, int $expires, string $path, string $key, string $ipRange = null, string $digest = 'sha1'): string
+    {
+        if (!function_exists('hash_hmac')) {
+            throw new \RuntimeException(sprintf('tempUrl requires hash extension enabled.'));
+        }
+
+        if ($ipRange) {
+            $message = sprintf("ip=%s\n%s\n%s\n%s", $ipRange, $method, $expires, $path);
+        } else {
+            $message = sprintf("%s\n%s\n%s", $method, $expires, $path);
+        }
+
+        $signature = hash_hmac($digest, $message, $key);
+
+        // sha512 requires prefixing signature
+        $signature = 'sha512' === $digest ? 'sha512:'.$signature : $signature;
+
+        return sprintf('%s?temp_url_sig=%s&temp_url_expires=%s', $path, $signature, $expires);
```

Comment:
Given the $path is not encoded, I'd suggest encoding the actual temporary url path

```suggestion
        return sprintf('%s?temp_url_sig=%s&temp_url_expires=%s', urlencode($path), $signature, $expires);
```